applications/nrf5340_audio/src/modules/sd_card.c

```diff
@@ -222,3 +230,72 @@ int sd_card_init(void)
 
 	return 0;
 }
+
+static volatile bool seg_read_started = false;
+static struct fs_file_t f_seg_read_entry;
+
+int sd_card_segment_read_open(char const *const filename, const char *path_to_file)
+{
+	char abs_path_name[PATH_MAX_LEN + 1] = SD_ROOT_PATH;
+	strcat(abs_path_name, path_to_file);
+	printk("abs path name:\t%s\n", abs_path_name);
+	int ret;
+
+	if (!sd_init_success) {
+		return -ENODEV;
+	}
+
+	if (strlen(filename) > CONFIG_FS_FATFS_MAX_LFN) {
+		LOG_ERR("Filename is too long");
+		return -FR_INVALID_NAME;
```

Comment:
Here we may need to discuss. The `FR_INVALID_NAME` you have chosen is a good decision, but this collides with the standard error defines.

Reply:
There is one called ENAMETOOLONG that can be used